masonry/src/widget/text_area.rs

```diff
@@ -709,7 +713,50 @@ impl<const EDITABLE: bool> Widget for TextArea<EDITABLE> {
             TextEvent::FocusChange(_) => {}
             TextEvent::Ime(e) => {
                 // TODO: Handle the cursor movement things from https://github.com/rust-windowing/winit/pull/3824
-                tracing::warn!(event = ?e, "TextArea doesn't accept IME");
+                let (fctx, lctx) = ctx.text_contexts();
+                // The text to submit as the TextChanged action. We do not send a TextChange action
+                // for the "virtual" preedit text.
+                let mut submit_text = None;
+                match e {
+                    winit::event::Ime::Disabled => {
+                        self.editor.transact(fctx, lctx, |txn| txn.clear_compose());
+                    }
+                    winit::event::Ime::Preedit(text, cursor) => {
+                        if text.is_empty() {
+                            self.editor.transact(fctx, lctx, |txn| txn.clear_compose());
+                        } else {
+                            if !self.editor.is_composing() && self.editor.selected_text().is_some()
+                            {
+                                // The IME has started composing. Delete the current selection and
+                                // send a TextChange event with the selection removed, but without
+                                // the composing preedit text.
+                                self.editor.transact(fctx, lctx, |txn| {
+                                    txn.delete_selection();
+                                });
+                                submit_text = Some(self.text().to_string());
+                            }
+
+                            self.editor
+                                .transact(fctx, lctx, |txn| txn.set_compose(text, *cursor));
+                        }
+                    }
+                    winit::event::Ime::Commit(text) => {
+                        self.editor
+                            .transact(fctx, lctx, |txn| txn.insert_or_replace_selection(text));
+                        submit_text = Some(self.text().to_string());
+                    }
+                    _ => {}
```

Comment:
```suggestion
                    winit::event::Ime::Enabled => {}
```
We know we don't need to handle `Enabled`, but we might need to handle some other event kinds.